### PR TITLE
JENKINS-54634 install evergreen in a dedicated VPC (fixed)

### PR DESCRIPTION
**Pull request by candrews**
Fixed problems in https://github.com/jenkins-infra/evergreen/pull/352
`EvergreenMasterSecurityGroup` and `EvergreenAgentSecurityGroup` did not have the `VpcId` property set (so they were being created in the default VPC)
`EC2EvergreenInstance` was missing the `SubnetId` property, so it was being created in the default VPC

All concerns expressed in https://github.com/jenkins-infra/evergreen/pull/396 have been addressed.

Also added an EIP so the IP address of the Evergreen instance doesn't change if the instance is recreated.

### CODE REVIEW

File: distribution/flavors/aws-ec2-cloud/CloudFormation/cloudformation-template.json [hunk 1]
```diff
@@ -22,9 +22,105 @@
...
       "ConstraintDescription": "Must be a valid IP CIDR range of the form x.x.x.x/x"
+    },
+    "VPCIPRange": {
+      "Description": "The private IP address range for allocating IPs within the VPC.",
+      "Type": "String",
+      "MinLength": "9",
+      "MaxLength": "18",
+      "Default": "10.0.0.0/16",
+      "AllowedPattern": "(\\d{1,3})\\.(\\d{1,3})\\.(\\d{1,3})\\.(\\d{1,3})/(\\d{1,2})",
+      "ConstraintDescription": "must be a valid IP CIDR range of the form x.x.x.x/x."
```
Comment (jsoref) [inline distribution/flavors/aws-ec2-cloud/CloudFormation/cloudformation-template.json]: Is this possible:
```suggestion
      "ConstraintDescription": "must be a valid IP CIDR range of the form `x.x.x.x/x`."
```

Comment (candrews) [reply distribution/flavors/aws-ec2-cloud/CloudFormation/cloudformation-template.json]: Certainly - I've updated this PR accordingly.

**Comment by candrews** [comment]
@jsoref can you please review this and hopefully merge it?

**Comment by candrews** [comment]
> lgtm (note: i'm not an evergreen person, so i don't know that my approval would count for anything)

Who can merge this?

**Comment by jsoref** [comment]
@rtyler

**Comment by candrews** [comment]
> LGTM

Awesome! Does that mean you can merge it? :crossed_fingers:

**Comment by mandie722** [comment]
Sorry, was waiting for the successful build and got distracted.

**Comment by candrews** [comment]
Thank you!

**Comment by candrews** [comment]
@mandie722 how does the `jenkins/evergreen:aws-ec2-cloud` image get updated?

I noticed that https://hub.docker.com/r/jenkins/evergreen/tags shows it being last updated 2 days ago, but I believe this commit should have caused it to be updated.

**Comment by mandie722** [comment]
Merging should trigger a new image, but the builds do take a really long time.

**Pull request merged**